### PR TITLE
Add cue position and timing to Trial table

### PR DESCRIPTION
**Pull request by weiglszonja**
Add more detailed information about cue position and timing for each trial

### CODE REVIEW

**Comment by bendichter** [comment]
@weiglszonja Could you store the times in seconds wrt the session start time?

**Comment by bendichter** [comment]
@weiglszonja is this ready for review?

File: tank_lab_to_nwb/convert_towers_task/virmenbehaviordatainterface.py [hunk 1]
```diff
@@ -118,49 +118,50 @@ def convert_data(self, nwbfile: NWBFile, metadata_dict: dict, stub_test: bool =
...
-            left_cue_table.add_column(
-                name='left_cue_position',
-                description='position of the left cues',
-                data=left_cue_positions_data,
-                index=left_cue_data_indices,
-            )
+            nwbfile.add_trial_column(name=cue_onset_table.name,
+                                     description=cue_onset_table.description,
+                                     index=cue_onset_table.left_cue_onset_index,
+                                     table=cue_onset_table)
```
Comment (weiglszonja) [inline tank_lab_to_nwb/convert_towers_task/virmenbehaviordatainterface.py]: @bendichter This solution produces a weird nwb file. Any suggestions how to add DynamicTable to Trial table? @CodyCBakerPhD 
I'd add `cueOffset` and `cuePosition` similarly once I figure out how to do it with `cueOnset`.

Comment (bendichter) [reply tank_lab_to_nwb/convert_towers_task/virmenbehaviordatainterface.py]: I don't think NaN-padding is the answer. How does the NWB file look weird?

Comment (weiglszonja) [reply tank_lab_to_nwb/convert_towers_task/virmenbehaviordatainterface.py]: without NaN padding I got an error that the length of the cue data doesn't match with the already existing trial column. The problem with the file is with `add_trial_column` it creates a table under `intervals/trials` with this `left_cue_onset_index` name, but when you open this table there is a whole duplication of file structure inside this table without the actual cue data.

**Comment by bendichter** [comment]
1. Do not create a new `DynamicTable`. Use the trials table that already exists.

**Comment by bendichter** [comment]
@weiglszonja try this

File: tank_lab_to_nwb/convert_towers_task/virmenbehaviordatainterface.py [hunk 2]
```diff
@@ -78,102 +79,137 @@ def convert_data(self, nwbfile: NWBFile, metadata_dict: dict, stub_test: bool =
...
-                SpatialSeries(
-                    name="SpatialSeries",
-                    data=H5DataIO(pos_data, compression="gzip"),
-                    reference_frame="unknown",
-                    conversion=0.01,
-                    resolution=np.nan,
-                    timestamps=H5DataIO(timestamps, compression="gzip")
-                )
+        if not Path(mat_file).is_file():
+            raise ValueError('{} not a file'.format(nat_file))
```
Comment (bendichter) [inline tank_lab_to_nwb/convert_towers_task/virmenbehaviordatainterface.py]:
```suggestion
            raise ValueError('{} not a file'.format(mat_file))
```

File: tank_lab_to_nwb/convert_towers_task/virmenbehaviordatainterface.py [hunk 3]
```diff
@@ -78,102 +79,137 @@ def convert_data(self, nwbfile: NWBFile, metadata_dict: dict, stub_test: bool =
...
+        right_padding = np.full((len(trial_starts) - len(right_cue_onsets)), np.nan)
+        right_cue_onsets = np.concatenate([right_cue_onsets, right_padding], axis=0)
+        right_cue_onset_data, right_cue_data_indices = create_indexed_array(right_cue_onsets)
+
+        cue_onset_table.add_column(
+            name='left_cue_onset',
+            description='onset times of left cues',
+            data=left_cue_onset_data,
+            index=left_cue_data_indices,
+        )
```
Comment (bendichter) [inline tank_lab_to_nwb/convert_towers_task/virmenbehaviordatainterface.py]:
```suggestion
```

**Comment by bendichter** [comment]
@weiglszonja let me know when this is ready to review

**Comment by weiglszonja** [comment]
@bendichter should I remove `cue_orientation` now that we have all the detailed information about cues?

**Comment by bendichter** [comment]
@weiglszonja yeah, let's remove it. Also, can you fix the merge conflicts?

**Comment by weiglszonja** [comment]
I have issues in `towersnwbconverter.py` after merging main into this branch. 
For: `super().get_metadata()` I get this error: `'NoneType' object has no attribute 'items'`
I updated `nwb-conversion-tools` though

**Pull request merged**